code/modules/mob/living/basic/bots/cleanbot/cleanbot.dm

```diff
@@ -0,0 +1,349 @@
+
+//Cleanbot
+/mob/living/basic/bot/cleanbot
+	name = "\improper Cleanbot"
+	desc = "A little cleaning robot, he looks so excited!"
+	icon = 'icons/mob/silicon/aibots.dmi'
+	icon_state = "cleanbot0"
+	pass_flags = PASSMOB | PASSFLAPS
+	density = FALSE
+	anchored = FALSE
+	health = 25
+	maxHealth = 25
+
+	maints_access_required = list(ACCESS_ROBOTICS, ACCESS_JANITOR)
+	radio_key = /obj/item/encryptionkey/headset_service
+	radio_channel = RADIO_CHANNEL_SERVICE
+	bot_type = CLEAN_BOT
+	hackables = "cleaning software"
+	additional_access = /datum/id_trim/job/janitor
+	greyscale_config = /datum/greyscale_config/buckets_cleanbot
+	possessed_message = "You are a cleanbot! Clean the station to the best of your ability!"
+	ai_controller = /datum/ai_controller/basic_controller/bot/cleanbot
+	///the bucket used to build us.
+	var/obj/item/reagent_containers/cup/bucket/build_bucket
+	///Flags indicating what kind of cleanables we should scan for to set as our target to clean.
+	///Options: CLEANBOT_CLEAN_BLOOD | CLEANBOT_CLEAN_TRASH | CLEANBOT_CLEAN_PESTS | CLEANBOT_CLEAN_DRAWINGS
+	var/janitor_mode_flags = CLEANBOT_CLEAN_BLOOD
+	///should other bots salute us?
+	var/comissioned = FALSE
+	///the base icon state, used in updating icons.
+	var/base_icon = "cleanbot"
+	/// if we have all the top titles, grant achievements to living mobs that gaze upon our cleanbot god
+	var/ascended = FALSE
+	///List of all stolen names the cleanbot currently has.
+	var/list/stolen_valor = list()
+	///Currently attached weapon, usually a knife.
+	var/obj/item/weapon
+	///our mop item
+	var/obj/item/mop/our_mop
+	///list of our officer titles
+	var/static/list/officers_titles = list(
+		JOB_CAPTAIN,
+		JOB_HEAD_OF_PERSONNEL,
+		JOB_HEAD_OF_SECURITY,
+		JOB_RESEARCH_DIRECTOR,
+	)
+	///job titles we can get
+	var/static/list/job_titles = list(
```

Comment:
feels like we should unit test that every job is in here

Reply:
this doesnt have all jobs. just a select few